cli/migrate.go

```diff
@@ -283,42 +321,31 @@ func publishEventWithOptions(ctx context_pkg.Context, envName string, modifiedSc
 		if err := env.HandleEvent(eventName, eventContext); err != nil {
 			log.Fatalf("Failed to handle event '%s': %s", eventName, err)
 		}
+
+		if err := migration.UnmarkSchema(s.ID); err != nil {
```

Comment:
This and `clearPendingSchemas` are the only real additions to this file. Other changes are simple mechanical refactorings so that they are low-risk. I'd love to add some unit tests here, but this would need to be changed a lot...